src/meltano/core/schedule_service.py

```diff
@@ -198,7 +198,7 @@ def default_start_date(self, session, extractor: str) -> datetime:  # noqa: ANN0
 
         # TODO: this coercion should be handled by the `kind` attribute
         # on the actual setting
-        if isinstance(start_date, date):
+        if isinstance(start_date, date) and not isinstance(start_date, datetime):
```

Comment:
turns out `datetime` is a subclass of `date` so the `if isinstance(start_date, datetime)` branch below was unreachable.

Reply:
> turns out `datetime` is a subclass of `date`

I feel like there are almost no beneficial uses of that property, whereas bugs resulting from it are easy to imagine 🫤